src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStepTest.java

```diff
@@ -90,6 +90,18 @@ public void writeExistingFile() throws Exception {
         j.assertLogContains("FileAlreadyExistsException", b);
     }
 
+    @Test
+    public void overwriteExistingFile() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "list");
+        p.setDefinition(new CpsFlowDefinition(
+                "node('slaves') {\n" +
+                        "  touch 'test.yml' \n" +
+                        "  writeYaml file: 'test.yml', overwrite: true, data: ['a', 'b', 'c'] \n" +
+                        "}",
+                true));
+        j.assertBuildStatusSuccess(p.scheduleBuild2(0));
```

Comment:
Also need to check that the file is actually overwritten and not just appended to (because I'm unsure how snakeyaml behaves in this situation). Probably in a separate test.

Reply:
This has probably nothing to do with snakeyaml since the code is just writing to a FilePath OutputStream. So the lib shouldn't be able to influence the file handling. But surely I'll add the test.

Reply:
Yes I think so too, just would like to make sure :)